### PR TITLE
Ad user model and specs

### PR DESCRIPTION
**Pull request by apalski**
PR for creating User model and User model specs.
Added error messages for user_spec.rb  to config/locales/en.yml.
Ran Rubocop and fixed violations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+require "./app/models/user"
```
Comment (elle) [inline spec/models/user_spec.rb]: Rails helper loads the Rails environment, thus line 2 is not needed.

If you weren't calling on `rails_helper`, then it would have been needed.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+describe User do
+  context "when all attributes are correct" do
+    it "is created" do
+      user = User.create(name: "John Doe",
```
Comment (elle) [inline spec/models/user_spec.rb]: Creating a user in the database is expensive.
Instead I would:
1. build a user and check if it is valid?
2. use shoulda-matchers

I would go for option 2

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      expect(user).to be_a User
+    end
+  end
+
+  context "when name not given" do
+    it "raises an error" do
+      user = User.create(email: "user@gmail.com")
+
+      expect { user.save! }
```
Comment (elle) [inline spec/models/user_spec.rb]: Why `.create` and then `.save!`

Once again, I would build first, call `user.valid?` and then check the error message.
But also, I would save me time and use shoulda-matchers

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        .to raise_error("Validation failed: Name must be entered")
```
Comment (elle) [inline spec/models/user_spec.rb]: Use translations in specs as well

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    end
+  end
+
+  context "when email not given" do
+    it "raises an error" do
+      user = User.create(name: "John Doe")
+
+      expect { user.save! }
+        .to raise_error("Validation failed: Email must be entered")
```
Comment (elle) [inline spec/models/user_spec.rb]: Same comments as with name

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    end
+  end
+
+  context "when name is not unique" do
+    it "raises an error" do
+      user = User.create(name: "John Doe",
+                         email: "user@gmail.com")
+      user.save!
```
Comment (elle) [inline spec/models/user_spec.rb]: You are creating and saving. Two calls to the database in one. Line 36 is not needed.
But shoulda-matchers can simplify this a lot.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (apalski) [reply spec/models/user_spec.rb]: Thanks Elle, ignore in all the next PRs as I will fix this across them all .

File: spec/models/user_spec.rb (hunk 3 above)
Comment (apalski) [reply spec/models/user_spec.rb]: Thanks Elle, again, ignore in next PRs as I will fix this across all specs

File: spec/models/user_spec.rb (hunk 4 above)
Comment (apalski) [reply spec/models/user_spec.rb]: Duh, of course! Thanks Elle - again ignore in next PRs as I will fix this across all specs

**Comment by apalski** [comment]
Hi @elle 
I think this PR is ready for review. Thanks again for your help!

File: config/locales/en.yml [hunk 7]
```diff
@@ -15,5 +15,21 @@ en:
...
+      unit: "$"
+
   titles:
     application: Budgetapp
+
+  activerecord:
+    errors:
+      messages:
+        blank: "must be entered"
+        taken: "already taken"
```
Comment (elle) [inline config/locales/en.yml]: Probably leftover from other PRs, which doesn't belong here.

File: spec/factories.rb [hunk 8]
```diff
@@ -1,2 +1,6 @@
 FactoryBot.define do
+  factory :user do
+    sequence(:name) { |n| "name#{n}" }
+    sequence(:email) { |n| "email#{n}" }
```
Comment (elle) [inline spec/factories.rb]: I would change this to a proper email format: i.e. `"email#{n}@example.com"`

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class User < ActiveRecord::Base
+
+  validates :name, :email, presence: true
+  validates :name, :email, uniqueness: true
```
Comment (elle) [inline app/models/user.rb]: You can do this lines in one declaration:
`validates :name, :email, presence: true, uniqueness: true`

Comment (elle) [reply app/models/user.rb]: What do yo think about validating email format?

File: config/routes.rb [hunk 10]
```diff
@@ -1,2 +1,6 @@
 Rails.application.routes.draw do
+  scope "/:locale", locale: /en/ do
+    resources :users, only: [:new]
+    resources :budgets, only: [:new]
+  end
```
Comment (elle) [inline config/routes.rb]: Same as comment from the other PR. This change doesn't belong in this PR

File: app/models/user.rb (hunk 9 above)
Comment (apalski) [reply app/models/user.rb]: Thanks Elle, I like the idea of validating the email format! I have added it to my project todo list!
